### DISCUSSION TITLE
Matlab direct interface

### DISCUSSION THREAD

**Discussion by gab5631** [snl-dakota/dakota#43] [Q&A]
I would like to employ the direct interface by which Dakota can directly drive Matlab.  However, I'm running a recent version of Matlab (in a Linux environment) and I see in the Dakota documentation that this capability is only supported up to Matlab 7.1 (R14SP3), per the following excerpt from Help:
"The Dakota/Matlab interface has been built and tested on 32-bit Linux with Matlab 7.0 (R14) and on 64-bit Linux with Matlab 7.1 (R14SP3)."

If possible, can you please provide additional instruction for how one can compile/install Dakota in such a way that ensures linkage with a more recent version of Matlab?

**Comment by briadam** [comment]
The Dakota Matlab direct interface hasn't been actively supported for some time. While prototyped some years ago and working at some level, we've never been able to reliably enable it, in part due to Matlab and Dakota nearly always depending on different versions of Boost shared libraries. My hunch is that a successful future Dakota/Matlab direct interface will have to depend on some kind of runtime plugin and/or inter-process communication, instead of build-time linking.

For now, I believe you're left with black-box Dakota/Matlab interfacing, which incurs the runtime overhead of exchanging variables/responses through files. An alternative you might look into is whether you could pair Dakota's Python direct interface with Matlab's Python bindings. I didn't see an easy path forward that way (that would counter file I/O inefficiencies _and_ Matlab startup time), but you may fare better.

Brian
